I've just pushed my first attempt at tests, I've got one working and one failing at the moment. I'm a bit stuck and could use some help getting the broken test to work.

@profclems any hints you can give me on using RegisterResponder correctly? For quick reference, here's what I've got:

```
	reg := &httpmock.Mocker{
		MatchURL: httpmock.PathAndQuerystring,
	}
	defer reg.Verify(t)

	reg.RegisterResponder("DELETE", "projects/owner%2Frepo/variables/TEST_VAR?filter%5Benvironment_scope%5D=%2A",
		httpmock.NewStringResponse(204, ""),
	)
```
I've tried different combinations of urlencoding and not urlencoding the various characters in the URL, but I've not come across a successful combination yet.

Here's what I'm currently getting when with the test fails:

```
$ go test -timeout 30s -run ^Test_deleteRun_project$ github.com/profclems/glab/commands/variable/delete
--- FAIL: Test_deleteRun_project (0.00s)
        Error Trace:    assertions.go:85
        		assertions.go:191
        		assertions.go:744
        		delete_test.go:117
        Error:		No error is expected but got Delete "https://gitlab.com/api/v4/projects/owner%2Frepo/variables/TEST_VAR?filter%5Benvironment_scope%5D=%2A": no registered stubs matched &{DELETE https://gitlab.com/api/v4/projects/owner%2Frepo/variables/TEST_VAR?filter%5Benvironment_scope%5D=%2A HTTP/1.1 1 1 map[Accept:[application/json] Private-Token:[] User-Agent:[GLab - GitLab CLI]] <nil> <nil> 0 [] false gitlab.com map[] map[] <nil> map[]   <nil> <nil> <nil> 0xc0000b6010}
        
    panic.go:617: 1 unmatched HTTP stubs
FAIL
FAIL	github.com/profclems/glab/commands/variable/delete	0.013s
FAIL
```

--- a/commands/variable/delete/delete_test.go
+++ b/commands/variable/delete/delete_test.go
@@ -87,33 +87,85 @@ func Test_NewCmdSet(t *testing.T) {
 	}
 }
 
-func Test_deleteRun_project(t *testing.T) {
+func Test_deleteRun(t *testing.T) {
 	reg := &httpmock.Mocker{
 		MatchURL: httpmock.PathAndQuerystring,
 	}
 	defer reg.Verify(t)
 
-	reg.RegisterResponder("DELETE", "projects/owner%2Frepo/variables/TEST_VAR?filter%5Benvironment_scope%5D=%2A",
-		httpmock.NewStringResponse(204, ""),
+	reg.RegisterResponder("DELETE", "/api/v4/projects/owner%2Frepo/variables/TEST_VAR?filter%5Benvironment_scope%5D=%2A",
+		httpmock.NewStringResponse(204, " "),
 	)
 
-	io, _, stdout, _ := iostreams.Test()
+	reg.RegisterResponder("DELETE", "/api/v4/projects/owner%2Frepo/variables/TEST_VAR?filter%5Benvironment_scope%5D=stage",
+		httpmock.NewStringResponse(204, " "),
+	)
 
-	opts := &DeleteOpts{
-		HTTPClient: func() (*gitlab.Client, error) {
-			a, _ := api.TestClient(&http.Client{Transport: reg}, "", "gitlab.com", false)
-			return a.Lab(), nil
-		},
-		BaseRepo: func() (glrepo.Interface, error) {
-			return glrepo.FromFullName("owner/repo")
-		},
-		IO:    io,
-		Key:   "TEST_VAR",
-		Scope: "*",
+	reg.RegisterResponder("DELETE", "/api/v4/groups/testGroup/variables/TEST_VAR",
+		httpmock.NewStringResponse(204, " "),
+	)
+
+	var httpClient = func() (*gitlab.Client, error) {
+		a, _ := api.TestClient(&http.Client{Transport: reg}, "", "gitlab.com", false)
+		return a.Lab(), nil
 	}
-	_, _ = opts.HTTPClient()
+	var baseRepo = func() (glrepo.Interface, error) {
+		return glrepo.FromFullName("owner/repo")
+	}
 
-	err := deleteRun(opts)
-	assert.NoError(t, err)
-	assert.Equal(t, stdout.String(), "✓ Deleted variable TEST_VAR with scope * for owner/repo\n")
+	tests := []struct {
+		name        string
+		opts        DeleteOpts
+		wantsErr    bool
+		wantsOutput string
+	}{
+		{
+			name: "delete project variable no scope",
+			opts: DeleteOpts{
+				HTTPClient: httpClient,
+				BaseRepo:   baseRepo,
+				Key:        "TEST_VAR",
+				Scope:      "*",
+			},
+			wantsErr:    false,
+			wantsOutput: "✓ Deleted variable TEST_VAR with scope * for owner/repo\n",
+		},
+		{
+			name: "delete project variable with stage scope",
+			opts: DeleteOpts{
+				HTTPClient: httpClient,
+				BaseRepo:   baseRepo,
+				Key:        "TEST_VAR",
+				Scope:      "stage",
+			},
+			wantsErr:    false,
+			wantsOutput: "✓ Deleted variable TEST_VAR with scope stage for owner/repo\n",
+		},
+		{
+			name: "delete group variable",
+			opts: DeleteOpts{
+				HTTPClient: httpClient,
+				BaseRepo:   baseRepo,
+				Key:        "TEST_VAR",
+				Scope:      "",
+				Group:      "testGroup",
+			},
+			wantsErr:    false,
+			wantsOutput: "✓ Deleted variable TEST_VAR for group testGroup\n",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, _ = tt.opts.HTTPClient()
+
+			io, _, stdout, _ := iostreams.Test()
+			tt.opts.IO = io
+			io.IsInTTY = false
+
+			err := deleteRun(&tt.opts)
+			assert.NoError(t, err)
+			assert.Equal(t, stdout.String(), tt.wantsOutput)
+		})
+	}
 }

